taar/recommenders/collaborative_recommender.py

```diff
@@ -81,10 +81,12 @@ def recommend(self, client_data, limit):
         distances = {}
         for addon in self.raw_item_matrix:
             # We don't really need to show the items we requested. They will always
-            # end up with the greatest score.
+            # end up with the greatest score. Also filter out legacy addons from the
+            # suggestions.
             hashed_id = str(addon.get("id"))
             if (hashed_id in installed_addons or
-                hashed_id not in self.addon_mapping):
+                hashed_id not in self.addon_mapping or
+                self.addon_mapping[hashed_id].get("isWebextension") is False):
```

Comment:
Shouldn't this condition include also the case where "isWebextension" doesn't exist?